scripts/code_style_rewrite_branch.py

```diff
@@ -0,0 +1,210 @@
+#!/usr/bin/env python3
+
+# if worktree not clean exit
+#
+# git checkout oldbranch
+# git rebase codestyle_change_commit~1
+#
+# for oldbranch_commit in oldbranch
+#
+#     git checkout -b workbranch
+#
+#     git reset --hard oldbranch_commit
+#     ./scripts/code_style.py --fix
+#     git reset --soft newbranch_commit
+#     git commit -a --reuse-message=$(oldbranch_commit)
+#     git rev-parse HEAD -> workbranch_commit_new
+#
+#     git checkout newbranch
+#     git cherry-pick workbranch_commit_new
```

Comment:
We could simplify this loop to:
```
git checkout newbranch
for oldbranch_commit in oldbranch
    git checkout oldbranch_commit -- .
    code_style.fix_style(git diff --staged --name-only *.[hc] tests/suites/*.function scripts/data_files/*.fmt)
    git commit --reuse-message=oldbranch_commit
```

Reply:
I think the `git diff` command needs to have `oldbranch_commit~  oldbranch_commit` specified otherwise we will commit all of the files that differ between the old branch and the new (which is all of them as they've all been restyled). Otherwise, that looks good I think.